### ISSUE TITLE
Finetuning ProtT5

### ISSUE THREAD

**Issue by BSharmi** [agemagician/ProtTrans#113]
Hi there!

Love the work! I am planning to fine tune ProtT5 for seq-to-seq generation and had a question on pre-training protT5. 

Was the full encoder-decoder model trained? If so then wouldn't training with MLM (excerpt from paper)
"Contrary to the original T5 model which masks spans of multiple tokens, we adopted BERT’s denoising objective to corrupt and reconstruct single tokens using a masking probability of 15%."
only update the weights of the encoder? What was the training procedure for ProtT5 so that I can load the weights and further fine tune? 
I was thinking to loading the model by 
````model = T5ForConditionalGeneration.from_pretrained("Rostlab/prot_t5_xl_uniref50")````

and follow https://github.com/huggingface/transformers/tree/main/examples/flax/language-modeling#t5-like-span-masked-language-modeling which was suggested in another issue as well.

Does it look right?

Thank you!

**Comment by BSharmi** [comment]
Thanks so much for the response!
Just one more follow up, I am working on preparing the datasets to fine tune ProtT5 and thinking to use the same protein sequence as input and output (unlike the google T5). Do you remember what was the input/output for ProtT5?

**Comment by mheinzinger** [comment]
So the input was an amino acid sequence e.g. "S E Q W <extra_id_0> N <extra_id_1> E <s>" and the output is "<pad> <extra_id_0> W <extra_id_1> C </s>"
The only thing you should double check are the special tokens. I keep mixing up where to put <s>, </s> & <pad>.
One indicates end-of-encoder-input, one start-of-decoder-output, and another end-of-decoder-output.
But this you can easily backtrace by using the ProtT5-tokenizer to tokenize an example (such as the one above) and it to return also labels (which you will need for training via teacher-forcing)

**Comment by BSharmi** [comment]
Awesome thanks!

**Comment by BSharmi** [comment]
Hey! I almost closed the issue and then wanted to confirm that from your example the reconstructed output will be 
`S E Q W W N N C E `?

**Comment by mheinzinger** [comment]
Update: https://github.com/agemagician/ProtTrans/issues/137#issuecomment-1817576165
